### PR TITLE
Adds support for bucket replication scaling

### PR DESCRIPTION
**Pull request by vbnrh**
Screenshot 2023-07-04 at 11 55 53 AM
Screenshot 2023-07-04 at 11 56 19 AM

Design: https://www.figma.com/proto/SAqm21mhaYFH6L01GIx9CO/bucketclass-replication?page-id=1%3A1555&type=design&node-id=124-2838&viewport=1158%2C229%2C0.21&scaling=min-zoom&starting-point-node-id=124%3A2763

### CODE REVIEW

File: packages/odf/components/mcg/CreateObjectBucketClaim.tsx [hunk 1]
```diff
@@ -315,33 +316,42 @@ export const CreateOBCForm: React.FC<CreateOBCFormProps> = (props) => {
...
+                  )}
+                  onChange={() => {
+                    // if this checkbox is disabled, then on this point we purge the contents of replication form data
+                    if (replicationEnabled)
+                      dispatch({
+                        type: 'setReplicationRuleFormData',
+                        data: [],
+                      });
+                    toggleReplication(!replicationEnabled);
+                  }}
```
Comment (bipuladh) [inline packages/odf/components/mcg/CreateObjectBucketClaim.tsx:337]: Move this out to a function.

File: packages/odf/components/mcg/replication-policy-form.tsx [hunk 2]
```diff
@@ -129,80 +132,114 @@ export const ReplicationPolicyForm: React.FC<ReplicationFormProps> = ({
     updateParentState(rules);
   };
 
-  const hasSyncDeletion = rules.some((rule) => rule.syncDeletion);
+  const disableSyncDeletions = (val: boolean) => {
```
Comment (bipuladh) [inline packages/odf/components/mcg/replication-policy-form.tsx:135]:
```suggestion
  const disableSyncDeletions = (val: boolean) => {
```
can we use a better name than `val`?

File: packages/odf/components/mcg/replication-policy-form.tsx [hunk 3]
```diff
@@ -129,80 +132,114 @@ export const ReplicationPolicyForm: React.FC<ReplicationFormProps> = ({
...
+            isChecked={eventLogsEnabled}
+            description={t(
+              'You must enable and configure object logging in the cloud environment of your choice'
             )}
-          </Grid>
-        ))}
+            onChange={(val: boolean) => {
+              toggleEventLogs(val);
+              disableSyncDeletions(val);
+            }}
```
Comment (bipuladh) [inline packages/odf/components/mcg/replication-policy-form.tsx:159]: Move it out to a function.

File: packages/odf/components/mcg/replication-policy-form.tsx [hunk 4]
```diff
@@ -129,80 +132,114 @@ export const ReplicationPolicyForm: React.FC<ReplicationFormProps> = ({
...
-              {t('Add rule')}
+              <PlusCircleIcon /> {t('Create new NamespaceStore')}
             </Button>
-          </FormGroup>
-        )}
+          </FlexItem>
+        </Flex>
+        <Stack hasGutter className="odf-mcg__form">
+          {rules.map((rule, index) => (
+            <StackItem>
```
Comment (bipuladh) [inline packages/odf/components/mcg/replication-policy-form.tsx:192]:
```suggestion
            <StackItem key={rule.id}>
```

File: packages/odf/components/mcg/create-obc.scss [hunk 5]
```diff
@@ -7,15 +7,19 @@
...
   
   &__remove-button {
-   margin-top: var(--pf-global--spacer--xl)
+   margin-top: var(--pf-global--spacer--xl);
+   display: inline
   }
   
   &__form {
-    margin-left: var(--pf-global--spacer--xl)
+
```
Comment (SanjalKatiyar) [inline packages/odf/components/mcg/create-obc.scss:15]: nits:
```suggestion
```

**Comment by SanjalKatiyar** [comment]
plz run `yarn i18n`

File: packages/odf/components/mcg/create-obc.scss (hunk 5 above)
Comment (vbnrh) [reply packages/odf/components/mcg/create-obc.scss]: done

File: packages/odf/components/mcg/replication-policy-form.tsx (hunk 4 above)
Comment (vbnrh) [reply packages/odf/components/mcg/replication-policy-form.tsx]: done

File: packages/odf/components/mcg/replication-policy-form.tsx (hunk 3 above)
Comment (vbnrh) [reply packages/odf/components/mcg/replication-policy-form.tsx]: done

File: packages/odf/components/mcg/replication-policy-form.tsx (hunk 2 above)
Comment (vbnrh) [reply packages/odf/components/mcg/replication-policy-form.tsx]: changed to isEventLogEnabled

File: packages/odf/components/mcg/CreateObjectBucketClaim.tsx (hunk 1 above)
Comment (vbnrh) [reply packages/odf/components/mcg/CreateObjectBucketClaim.tsx]: done

File: packages/odf/components/mcg/CreateObjectBucketClaim.tsx [hunk 6]
```diff
@@ -233,6 +234,16 @@ export const CreateOBCForm: React.FC<CreateOBCFormProps> = (props) => {
...
 
+  const onChangeReplication = () => {
+    // if this checkbox is disabled, then on this point we purge the contents of replication form data
+    if (replicationEnabled)
+      dispatch({
+        type: 'setReplicationRuleFormData',
+        data: [],
+      });
+    toggleReplication(!replicationEnabled);
+  };
```
Comment (bipuladh) [inline packages/odf/components/mcg/CreateObjectBucketClaim.tsx:245]: You need to wrap it in a useCallback.

File: packages/odf/components/mcg/replication-policy-form.tsx [hunk 7]
```diff
@@ -129,80 +132,116 @@ export const ReplicationPolicyForm: React.FC<ReplicationFormProps> = ({
...
 
-  const hasSyncDeletion = rules.some((rule) => rule.syncDeletion);
+  const disableSyncDeletions = (isEventLogEnabled: boolean) => {
+    const newRules: Array<Rule> = rules.map((rule) => ({
+      ...rule,
+      syncDeletion: isEventLogEnabled ? rule.syncDeletion : false,
+    }));
+    setRules(newRules);
+    updateParentState(newRules);
+  };
```
Comment (bipuladh) [inline packages/odf/components/mcg/replication-policy-form.tsx:142]: Wrap it in a useCallback

File: packages/odf/components/mcg/CreateObjectBucketClaim.tsx (hunk 6 above)
Comment (vbnrh) [reply packages/odf/components/mcg/CreateObjectBucketClaim.tsx]: done

File: packages/odf/components/mcg/replication-policy-form.tsx (hunk 7 above)
Comment (vbnrh) [reply packages/odf/components/mcg/replication-policy-form.tsx]: done

**Comment by vbnrh** [comment]
Screenshot 2023-07-05 at 3 30 43 PM

**Review by bipuladh — APPROVED** [review]

**Comment by bipuladh** [comment]
/lgtm cancel
@SanjalKatiyar to add the final lgtm

**Pull request merged**